Vienna/Sources/Main window/CustomWKWebView.swift

```diff
@@ -173,6 +178,50 @@ class CustomWKWebView: WKWebView {
         }
         return text
     }
+
+    // MARK: Text zoom
+
+    var canMakeTextLarger: Bool {
+        guard responds(to: #selector(getter: _supportsTextZoom)),
+              responds(to: #selector(getter: _textZoomFactor)),
+              _supportsTextZoom
+        else {
+            return false
+        }
+
+        return Float(_textZoomFactor) < 3.0
+    }
+
+    var canMakeTextSmaller: Bool {
+        guard responds(to: #selector(getter: _supportsTextZoom)),
+              responds(to: #selector(getter: _textZoomFactor)),
+              _supportsTextZoom
+        else {
+            return false
+        }
+
+        return Float(_textZoomFactor) > 0.5
+    }
+
+    @objc
+    func makeTextLarger(_ sender: Any) {
+        guard canMakeTextLarger
+        else {
+            return
+        }
+
+        _textZoomFactor += 0.1
+    }
```

Comment:
I think you need to retain the selector check, because you are accessing the private setter here (`canMakeTextLarger` only uses the getter).
```swift
func makeTextLarger(_ sender: Any) {
    guard responds(to: #selector(setter: _textZoomFactor)),
          canMakeTextBigger
    else {
        return
    }

    _textZoomFactor += 0.1
}
```

Reply:
Done, just note that for sake of name consistency, mine is named `canMakeTextLarger` rather can `canMakeTextBigger`

Reply:
By the way, why not just testing the setter in `canMakeTextLarger` ?